central/vulnmgmt/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,207 @@
+		err := s.deployments.UpsertDeployment(upsertCtx, deployment)
+		s.Require().NoError(err)
+		for _, image := range fixtures.DeploymentImages() {
+			err = s.images.UpsertImage(upsertCtx, image)
+			s.Require().NoError(err)
+		}
+	}
+}
+
+func (s *servicePostgresTestSuite) receiveWorkloads(client v1.VulnMgmtServiceClient,
+	request *v1.VulnMgmtExportWorkloadsRequest,
+) []*v1.VulnMgmtExportWorkloadsResponse {
+	out, err := client.VulnMgmtExportWorkloads(s.ctx, request)
+	s.Require().NoError(err)
+	var results []*v1.VulnMgmtExportWorkloadsResponse
+	for {
+		chunk, err := out.Recv()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		results = append(results, chunk)
+	}
+	return results
+}
+
+func (s *servicePostgresTestSuite) TestExport() {
+	cases := []struct {
+		name        string
+		deployments []*storage.Deployment
+		query       string
+		expected    []*storage.Deployment
+	}{
+		{
+			name: "zero deployments",
+		},
+		{
+			name: "one deployment no query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+			},
+			expected: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+			},
+		},
+		{
+			name: "multiple deployments no query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+				s.createDeployment(fixtureconsts.Deployment2),
+				s.createDeployment(fixtureconsts.Deployment3),
+			},
+			expected: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+				s.createDeployment(fixtureconsts.Deployment2),
+				s.createDeployment(fixtureconsts.Deployment3),
+			},
+		},
+		{
+			name: "multiple deployments with query",
+			deployments: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment1),
+				s.createDeployment(fixtureconsts.Deployment2),
+				s.createDeployment(fixtureconsts.Deployment3),
+			},
+			query: fmt.Sprintf("Deployment ID:%s", fixtureconsts.Deployment2),
+			expected: []*storage.Deployment{
+				s.createDeployment(fixtureconsts.Deployment2),
+			},
+		},
+	}
+	for _, c := range cases {
+		s.T().Run(c.name, func(t *testing.T) {
+			s.upsertDeployments(c.deployments)
+
+			request := &v1.VulnMgmtExportWorkloadsRequest{Timeout: 5, Query: c.query}
+			conn, closeFunc := s.createGRPCWorkloadsService(s.ctx)
+			defer closeFunc()
+			client := v1.NewVulnMgmtServiceClient(conn)
+			results := s.receiveWorkloads(client, request)
+
+			// The images are the same for all deployments to simplify the assertions.
+			expectedImages := fixtures.DeploymentImages()
+			expectedImageIDs := []string{expectedImages[0].GetId(), expectedImages[1].GetId()}
+
+			// We cannot perform a full assert on the response because it contains variable data
+			// and timestamps.
+			s.Require().Len(results, len(c.expected))
+			for i := range results {
+				s.Assert().Contains(c.deployments, results[i].Deployment)
+				s.Require().Len(results[i].Images, len(expectedImages))
```

Comment:
Since all deployments are supposed to have the same images, how about pulling the image IDs from the results into a list and perform an `assert.ElementMatch` on the retrieved list ?